src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -140,24 +155,310 @@ IMF_STD_ATTRIBUTE_DEF (longitude, Longitude, float)
+
 //
 // isoSpeed -- the ISO speed of the film or image sensor
 // that was used to record the image
 //
 
 IMF_STD_ATTRIBUTE_DEF (isoSpeed, IsoSpeed, float)
 
+//
 // exposure -- exposure time, in seconds
+//
 
 IMF_STD_ATTRIBUTE_DEF (expTime, ExpTime, float)
 
 //
-// aperture -- the camera's lens aperture, in f-stops (focal length
-// of the lens divided by the diameter of the iris opening)
+// shutterAngle -- shutter angle, in degrees
+//
+
+IMF_STD_ATTRIBUTE_DEF (shutterAngle, ShutterAngle, float)
+
+//
+// captureRate -- capture rate, in frames per second, of the image
+// sequence to which the image belongs, represented as a rational
+// number
+// 
+// For variable frame rates, time-lapse photography, etc. the capture
+// rate r is calculated as
+//
+//   r = 1 / (tN - tNm1)
+//
+// where tn is the time, in seconds, of the center of frame N's 
+// exposure interval, and tNm1 is the time, in seconds, of the center
+// of frame N-1's exposure interval.
+//
+// Both the numerator and denominator of r must be strictly positive.
+//
+
+IMF_STD_ATTRIBUTE_DEF (captureRate, CaptureRate, Rational)
+
+//
+// timecodeRate -- the timecode rate associated with the playback of
+// the image sequence to which the image belongs, specified in
+// timecodes per second
+//
+
+IMF_STD_ATTRIBUTE_DEF (timecodeRate, TimecodeRate, int)
+
+//
+// lensMake -- manufacturer or vendor of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensMake, LensMake, std::string)
+
+//
+// lensModel -- model name or model number of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensModel, LensModel, std::string)
+
+//
+// lensSerialNumber -- serial number of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length. Note that despite the name, the
+// string can include non-digits as well as digits.
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensSerialNumber, LensSerialNumber, std::string)
+
+//
+// lensFirmwareVersion -- firmware version of the lens
+//
+// If present, the value should contain only printable characters
+// and have a nonzero length.
+//
+
+IMF_STD_ATTRIBUTE_DEF (lensFirmwareVersion, LensFirmwareVersion, std::string)
+
+//
+// nominalFocalLength -- number printed on barrel of a prime lens, or
+// number next to index mark on a zoom lens, in units of millimeters.
+//
+// Nominal focal length is appropriate for asset tracking of lenses (e.g.
+// a camera rental house catalogs its lens stock by nominal focal length).
+//
+
+IMF_STD_ATTRIBUTE_DEF (nominalFocalLength, NominalFocalLength, float)
+
+//
+// pinholeFocalLength -- in the simplest model of image formation, the
```

Comment:
in all of these focalLength attributes, how are we to disambiguate which wins in the case that they are inconsistent?

Would we interpret values of 0.f to mean "not recorded" if the attribute is present?

If a DP was recording the focalLength as marked on a camera (e.g. 58mm) or marked on the focal ring on a zoom (108mm) which of the focal length attributes should hold that "painted on the hardware" value?

Reply:
There is no 'wins' as they are not redundant. You cannot derive any one of these from one or both of the others.

I suggest we mandate that the attribute value be strictly positive.

As for the example in your final paragraph, that's nominal focal length.

Reply:
Could we make a note for the nomenclaturally naive that the number painted on the dial is the nominal focal length? 

I suppose there are no diverging lenses we'd care about, such as some weird eye glass prescription, so a strictly positive restriction makes sense.

Reply:
understood as "as recorded"